superchain/extra/addresses/addresses.json

```diff
@@ -42,7 +42,6 @@
     "Proposer": "0xf8Ed03b83c15aa3d0b52095F3c9971225948B777",
     "ProxyAdmin": "0xd60a706Bf6108F090d055787B9B353FA7EEE1355",
     "ProxyAdminOwner": "0x4c4526C8C55c4e1a6F2aaf91e37f07c97786e0f5",
-    "SuperchainConfig": "0xe5b3692266FF4Ab8A96A9C7Da6EeEe532CCc7916",
```

Comment:
What is the intent of `addresses.json` and who should consume it? I would expect that it moves in the opposite direction to serve as a flat, comprehensive list of all contracts for that chain, to avoid needing view a few different files to collect all addresses

It seems like currently for chains not using the standard `SuperchainConfig`, there is no way to read their `SuperchainConfig` from the registry? In which case we could look up their `SuperchainConfig` during codegen and that way the flat codegen'd `addresses.json` contains all addresses a suprchain registry consumer would need

Reply:
I believe it is there for the convenience of downstream applications / other users. Those apps may not wish to discover addresses on chain -- it's worse than just going over multiple files since the `SuperchainConfig` can actually vary per chain and not match the standard superchain address (we only validate standard chains are compliant). 

I think the codegen here is a bit of a distraction, it will contain the same information as the individual TOML files. It is just that some of the individual TOML files are missing the `SuperchainConfig` address.

It sounds like the better change might be to backfill missing addresses then, and add a validation check for all chains that the in chain references resolve to the ones in the registry. Would you be in favor of this @sebastianst @mds1 ? I'll push this back to a draft PR until we can align.